dev/PowerNotifications/PowerNotifications.h

```diff
@@ -281,71 +235,103 @@ namespace winrt::Microsoft::Windows::System::Power
 
             event_token AddCallback(PowerFunctionDetails fn, const PowerEventHandler& handler)
             {
-                auto& eventObj{ fn.event() };
-                std::scoped_lock<std::mutex> lock(m_mutex);
-                if (!RegisteredForEvents(eventObj))
+                try
+                {
+                    PowerNotifications::AddCallbackEvent(fn.name.c_str());
+                    auto& eventObj{ fn.event() };
+                    std::scoped_lock<std::mutex> lock(m_mutex);
+                    if (!RegisteredForEvents(eventObj))
+                    {
+                        PowerNotifications::RegisterEvent(fn.name.c_str());
```

Comment:
Readability feedback: because of the way things are named, it really sounds like RegisteredForEvents and RegisterEvent are related to each other; it's not clear on reading this that RegisteredForEvents is functional, but RegisterEvent is telemetry. Might consider massaging the naming.

Reply:
Maybe TraceRegister instead of RegisterEvent? I think verb naming fits better than noun naming here anyway.

Reply:
Great point. Changed.

Reply:
Oh yeah, changing the class name was also a good call.